### ISSUE TITLE
Dynamic Color in Gradient background not working

### ISSUE THREAD

**Issue by satanasdiabolo** [elementor/elementor#10885]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

## Description

When creating an element with a gradient background, with dynamic colours. The gradient is displayed on the editor but not on the final post.
*Editor*
![Capture du 2020-03-18 17-29-24](https://user-images.githubusercontent.com/1850524/76983647-0cb07080-693e-11ea-9772-c1fe2b6ef747.png)
*Result*
![image](https://user-images.githubusercontent.com/1850524/76983776-3f5a6900-693e-11ea-9326-51a43c57eb7e.png)

With WP_DEBUG set to true, I have the following errors:
* Notice: Undefined index: background_color in .../wp-content/plugins/elementor/core/files/css/base.php on line 668

## Steps to reproduce

* Download the test plugin and install it [test.zip](https://github.com/elementor/elementor/files/4350097/test.zip)
* Edit any post
* Select a section or a column
* Go to Style > Background > Gradient
* For the two colors select the dynamic color
* Choose a color from the color picker
* See the result in the editor
* Save the post
* Consult the single page of the post

## Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a default WordPress theme active.
- [X] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
					
== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.29 (Ubuntu)
	MySQL version: (Ubuntu) v5.7.29
	PHP Version: 7.2.24-0ubuntu0.18.04.3
	PHP Max Input Vars: 1000
	PHP Max Post Size: 8M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: There are some writing permissions issues with the following directories/files:
		 - WordPress root uploads directory
		 - WordPress root directory
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.3.2
	Site URL: http://127.0.0.1/test
	Home URL: http://127.0.0.1/test
	WP Multisite: No
	Max Upload Size: 2 MB
	Memory limit: 256M
	Permalink Structure: /index.php/%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Active

== Theme ==
	Name: Twenty Twenty
	Version: 1.1
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:74.0) Gecko/20100101 Firefox/74.0

== Active Plugins ==
	Dynamic colors test
		Version: 0.0.1
		Author: satanasdiabolo

	Elementor
		Version: 2.9.6
		Author: Elementor.com

	Elementor Pro
		Version: 2.9.1
		Author: Elementor.com

== Elements Usage ==
	
	wp-post : 1
		text-editor : 1

				
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

**Comment by elasticeye** [comment]
Have the same problem.

**Comment by rodolphebertozzo** [comment]
#10890

**Comment by rodolphebertozzo** [comment]
+1

**Comment by shilo-ey** [comment]
Thanks for reporting!

We will look into that and update this thread soon.

**Comment by dalt45** [comment]
I have the same problem tried with JetEngine and ACF for the custom color but instead of gradient I get a single color background. Is there still any solution to this?

**Comment by mshehrozsajjad** [comment]
I have the same issue and I have been facing it since the release of dynamic field features. Any idea how long it will take to fix it? Or any workaround someone found?

**Comment by PabloLombriz** [comment]
Same problem here with dynamic color. Working with Jet Engine.

**Comment by kino89** [comment]
Same problem here with dynamic color gradient background and ACF PRO. Elementor set in frontend only the first dynamic color as simple color background.

**Comment by mclendening** [comment]
Same problem using ACF, Toolset, and others, any ETA on a fix?

**Comment by getunrealtoday** [comment]
This is an issue for me as well. Is there any update on a fix. Surely it can't be that hard?

**Comment by mclendening** [comment]
Here's a fix I put into place that works perfectly. The use case was to give art/designers the ability to dial in the gradient overlay for a movie streaming site.
Components to make this work:
ACF Pro (not free, maybe the free version will work)
ACF RGBA plugin (free)
Snippets plugin (free)

Steps to make it work:
Create ACF field groups to hold the gradient values that will be used in a Snippet shortcode
Create Snippet shortcode that outputs CSS
Add Shortcode created in Snippets plugin to Custom Attributes of the desired Elementor Section (Advanced Tab)

ACF Field Group in custom post type:

![RGBA ACF Field Group](https://user-images.githubusercontent.com/451641/107247030-df116180-69fe-11eb-983a-f3580f57d895.png)


Snippet:
![RGBA Snippet](https://user-images.githubusercontent.com/451641/107245923-b341ac00-69fd-11eb-83c7-07718e51aaeb.png)

Snippet Shortcode placement in Elementor Section:

![RGBA Shorctode Advanced Attributes](https://user-images.githubusercontent.com/451641/107246337-28ad7c80-69fe-11eb-9db9-71425d82d4d4.png)

The output of the Snippet shortcode:

![RGBA Shortcode Output](https://user-images.githubusercontent.com/451641/107246714-86da5f80-69fe-11eb-8d64-4ba96766043f.png)

**Comment by tomhargreaves** [comment]
Any update for an ETA on when this will be resolved? The issue was discovered almost a year ago, with many Elementor releases since, and still with no fix.

**Comment by nillde** [comment]
same issue here...any progress?

**Comment by TomBigsby** [comment]
Here's another fix, in the same idea than **mclendening** with no plugin and less complex : 
- JetEngine > create a textarea for the custom gradient
- Post type > write the gradient HTML code in the textarea `<style>.bloc-gradient {background: linear-gradient(0deg, #FFBE0B, #FFBE0B00)}</style>` or with rgba colors
- Elementor > add a dynamic field anywhere in the page (it will be invisible)
![image](https://user-images.githubusercontent.com/20105332/116219241-af473f80-a74b-11eb-920c-b800a7b34bf7.png)
- Dynamic field > Content > Source : meta-data / Meta field : your custom gradient.
![image](https://user-images.githubusercontent.com/20105332/116220821-511b5c00-a74d-11eb-8a36-f8604c67ac15.png)

**Comment by paulsto** [comment]
Adding to the list of people whos design is being blocked by this over one year old bug

**Comment by saltadmindev** [comment]
Still the issue persists, Ive also logged a support request but so far no reply

**Comment by saltadmindev** [comment]
**This is the reply ive had from Elementor support**

Thank you for contacting us.

This is an ongoing bug and is already reviewed by our development team.

Please note that fixing bugs requires core code changes which may take some time to solve.

Currently, we do not have an exact ETA for when this issue will be solved, but once the bug is fixed, we will update the status in our changelog page: https://elementor.com/pro/changelog/

Let me know if you need any additional assistance.

Sincerely 
Rizwan Mirza
Tier 2 Technical Support

**Comment by adeelwebmaster** [comment]
**Same issue elementor support did not update the issue in the new version also.**

**Comment by debihazelden** [comment]
+ 1 for background gradients using dynamic field colour picker

**Comment by gaplands** [comment]
+1

**Comment by gaplands** [comment]
> Here's another fix, in the same idea than **mclendening** with no plugin and less complex :
> 
> * JetEngine > create a textarea for the custom gradient
> * Post type > write the gradient HTML code in the textarea `<style>.bloc-gradient {background: linear-gradient(0deg, #FFBE0B, #FFBE0B00)}</style>` or with rgba colors
> * Elementor > add a dynamic field anywhere in the page (it will be invisible)
> 
> <img alt="" width="150" height="93" src="https://user-images.githubusercontent.com/20105332/116219241-af473f80-a74b-11eb-920c-b800a7b34bf7.png">
> 
> - Dynamic field > Content > Source : meta-data / Meta field : your custom gradient.
> <img alt="" width="322" height="140" src="https://user-images.githubusercontent.com/20105332/116220821-511b5c00-a74d-11eb-8a36-f8604c67ac15.png">

This didn't work for me.

**Comment by sbc640964** [comment]
@shilo-ey 
Not planning to address the issue?

**Comment by AJdigitaldesigns** [comment]
Issue still persists. 2 Years after initial log.

**Comment by momo-fr** [comment]
Not solution actually??? It doesn't seem a problem for the devs but for the users YES!!

**Comment by WebToStep** [comment]
Странно что решения проблемы досих пор нет, ведь ключевая роль elementor это удобство пользователя. А здесь приходится устраивать танцы с бубном 🤦‍♂️

**Comment by rafaeljuliani** [comment]
Still doesn't work...

**Comment by rafaeljuliani** [comment]
It would be better to disable the dynamic colors option for the gradients.

**Comment by TieLove333** [comment]
+1 on this. The color picker field does not work anywhere with Elementor. Not sure if doable but would be nice if elementor allowed a 'text' custom field to provide the color hex code in the color picker.

**Comment by anow2** [comment]
I was told that this would be fixed "soon"

2 months later, I follow up, they send me to github.

And of course, I find out its a bug that's been around for > 2 years...

**Comment by vongoh** [comment]
Same issue here - our product template design is based around pulling in dynamic colors behind products.  

The dynamic color is not displayed on the front end,  whether on 'Background' or 'Background Overlay' - same broken behavior.

Looks like there is no progress on solving this,  so time to re strategize the design ....

**Comment by ronaldtveen** [comment]
The odd thing is that it actually works in the Elementor back-end, but on the frontend the CSS is actually changed to a static color.

This is how I've set it up, hopefully it triggers a dev to take second look into the matter.

1. In ACF I've created a color picker *start* and *end* color:
    https://res.cloudinary.com/comsi-rtv/image/upload/v1678864610/ACF/acf-settings-gradient_start-end.png
    https://res.cloudinary.com/comsi-rtv/image/upload/v1678864610/ACF/acf-post-gradient.png
2. In Elementor I've set the section background to gradient, using the *start* and *end* colors from dynamic ACF color fields
    https://res.cloudinary.com/comsi-rtv/image/upload/v1678864610/ACF/elementor-gradient-settings.png
    As you can here the color gradient is actually applied in the Elementor preview window.
3. When saving and viewing it on the frontend however, the gradient is removed and it falls back to the static *start* color
    https://res.cloudinary.com/comsi-rtv/image/upload/v1678864610/ACF/elementor-gradient-removed-on-frontend.jpg
    In the screenshot above there's a little pop-up I've added to compare it to when you manually set the gradient without the dynamic field.

**Comment by rasmuskernen** [comment]
A quick fix that worked well for my use case where i only needed one of the 2 gradient colours to be dynamic was putting a static background with the dynamic colour then the none dynamic colour as an overlay gradient with the other colour for gradient overlay being 0 opacity to get the gradient effect.

Maybe a nobrainer fix, but would have helped me if i saw it here earlier ;)

**Comment by cazper57** [comment]
+1 April 2023 :(

**Comment by w3b-beweb** [comment]
+1 :(((

**Comment by unbrandeduser** [comment]
+1 - May 2023 - Maybe Elementor could ask their new integrated AI tool to fix this problem

**Comment by arulpatmose** [comment]
> Here's a fix I put into place that works perfectly. The use case was to give art/designers the ability to dial in the gradient overlay for a movie streaming site. Components to make this work: ACF Pro (not free, maybe the free version will work) ACF RGBA plugin (free) Snippets plugin (free)
> 
> Steps to make it work: Create ACF field groups to hold the gradient values that will be used in a Snippet shortcode Create Snippet shortcode that outputs CSS Add Shortcode created in Snippets plugin to Custom Attributes of the desired Elementor Section (Advanced Tab)
> 
> ACF Field Group in custom post type:
> 
> ![RGBA ACF Field Group](https://user-images.githubusercontent.com/451641/107247030-df116180-69fe-11eb-983a-f3580f57d895.png)
> 
> Snippet: ![RGBA Snippet](https://user-images.githubusercontent.com/451641/107245923-b341ac00-69fd-11eb-83c7-07718e51aaeb.png)
> 
> Snippet Shortcode placement in Elementor Section:
> 
> ![RGBA Shorctode Advanced Attributes](https://user-images.githubusercontent.com/451641/107246337-28ad7c80-69fe-11eb-9db9-71425d82d4d4.png)
> 
> The output of the Snippet shortcode:
> 
> ![RGBA Shortcode Output](https://user-images.githubusercontent.com/451641/107246714-86da5f80-69fe-11eb-8d64-4ba96766043f.png)

This solution worked for me with some customization. Hope we will get a official solution for this in future!

Thanks @mclendening

**Comment by aldoalberto** [comment]
> Here's a fix I put into place that works perfectly. The use case was to give art/designers the ability to dial in the gradient overlay for a movie streaming site. Components to make this work: ACF Pro (not free, maybe the free version will work) ACF RGBA plugin (free) Snippets plugin (free)
> 
> Steps to make it work: Create ACF field groups to hold the gradient values that will be used in a Snippet shortcode Create Snippet shortcode that outputs CSS Add Shortcode created in Snippets plugin to Custom Attributes of the desired Elementor Section (Advanced Tab)
> 
> ACF Field Group in custom post type:
> 
> ![RGBA ACF Field Group](https://user-images.githubusercontent.com/451641/107247030-df116180-69fe-11eb-983a-f3580f57d895.png)
> 
> Snippet: ![RGBA Snippet](https://user-images.githubusercontent.com/451641/107245923-b341ac00-69fd-11eb-83c7-07718e51aaeb.png)
> 
> Snippet Shortcode placement in Elementor Section:
> 
> ![RGBA Shorctode Advanced Attributes](https://user-images.githubusercontent.com/451641/107246337-28ad7c80-69fe-11eb-9db9-71425d82d4d4.png)
> 
> The output of the Snippet shortcode:
> 
> ![RGBA Shortcode Output](https://user-images.githubusercontent.com/451641/107246714-86da5f80-69fe-11eb-8d64-4ba96766043f.png)

Disculpe alguien sabe si sigue funcionando ete metodo, lo probe pero nome funciono gracias

**Comment by aldoalberto** [comment]
> Thanks @mclendening ♥️
> 
> Your solution worked for me with some customization. Hope we will get an official fix for this issue in future!

Buen dia podrias ayudarme, intente hacer esto pero no me salio de casualidad tengas algun video explicando estos pasos saludos cordiales

**Comment by aldoalberto** [comment]
> 

Buen dia podrias compartir tu codigo personalizado podria ser de gran ayuda

**Comment by Segmant** [comment]
**This is the eventual answer i got from Elementor Support**

Hello there,

Thank you for the details.

After further inspection, I can confirm the issue is related to our plugin. This behavior has already been reported to our development team for further examination. Our developers are aware of this bug and are working on fixing the issue in future updates, but unfortunately, I don’t have a precise ETA of when the fix will be released.

I cannot tell you the exact ETA on this matter, but what you can do is pay attention to our future update changelogs, which can be found here:[ https://elementor.com/help/changelog-of-elementor/](https://elementor.com/help/changelog-of-elementor/) to track when the fix is released.

We sincerely apologize for any misunderstanding or frustration possibly caused by this issue. We are working hard on improving our product so our customers can have the best possible user experience. Thank you for understanding!

For any other questions/issues, don't hesitate to create a new ticket, and our support team will gladly assist you!

Have a great day!

Best regards,
​
Danica Protic

**Comment by kaskad88** [comment]
+1

**Comment by greenstreetdesign** [comment]
+1 

Client wants to choose dynamic colour gradients for hero sections based on the product they are featuring. Only the first colour displays – works perfectly in the loop editor. 

I notice there's an extremely subtle shifting of colour saturation when mouseover on any links on the page; maybe this is part of the bug? 

Gradients (or lack thereof) in the right column of top hero section here:

https://store.lqa.com

**Comment by simonmilz** [comment]
+1

**Comment by richtiggut** [comment]
+1

**Comment by ale92ml** [comment]
I've been through the same issue and I've just created a simple solution.
Add this code to "functions.php". I recommend you to install "Hello Elementor Child" before adding the code.
It's in spanish, but basically I have a custom post type created with ACF called "libro" and two fields called "color_del_libro" and "color_del_libro_degradado", which are 2 different colors to compose the gradient.
Customize according your needs.
I don't know if it's the best way to solve it but it works.

```php
function obtener_color_libro() {
   
    if (is_single() && get_post_type() == 'libro') {
        $post_id = get_the_ID();
        $color_del_libro = get_field('color_del_libro', $post_id); //replace with the proper name field
        $color_del_libro_degradado = get_field('color_del_libro_degradado', $post_id); //replace with the proper name field

        if ($color_del_libro && $color_del_libro_degradado) {
            return array('color_del_libro' => $color_del_libro, 'color_del_libro_degradado' => $color_del_libro_degradado);
        }
    }

   //if nothing is found then just prints a simple black background as default
    return array('color_del_libro' => '#000000', 'color_del_libro_degradado' => '#000000');

}

function aplicar_color_libro() {
    if (is_single() && get_post_type() == 'libro') {
        $colores = obtener_color_libro();
        echo "<style>
                .contenedorlibro {
                    background-color: transparent;
                    background-image: linear-gradient(360deg, {$colores['color_del_libro']} 0%, {$colores['color_del_libro_degradado']} 100%);
                }
              </style>";
    }
}
add_action('wp_head', 'aplicar_color_libro');
``` 


If you need it to be more generic, or for the entire website, remove the custom post type checker: 
  if (is_single() && get_post_type() == 'libro') {

and replace...

echo "<style>
                .contenedorlibro {
                    background-color: transparent;
                    background-image: linear-gradient(360deg, {$colores['color_del_libro']} 0%, {$colores['color_del_libro_degradado']} 100%);
                }
              </style>";

with:

echo "<style>
        :root {
            --color-libro: {$colores['color_del_libro']};
            --color-libro-degradado: {$colores['color_del_libro_degradado']};
        }
      </style>";

Then, in elementor add these lines to the css code (if you have free version then add an HTML widget and add <style>CODE</style>"

.yourwrapper {
  background-color: transparent;
  background-image: linear-gradient(360deg, var(---color-libro) 0%, var(--color-libro-degradado) 100%);
}

Customize according your needs. And if it is not working, just add !important by the end of the line.
Also, you can use these custom variable for all the website.

I hope it works for you!

**Comment by BryBusiness** [comment]
+1 Same issue here!

**Comment by Marceloolz** [comment]
Hello! My problem was putting a dynamic color.

I managed to solve it without extra plugins, just doing it as follows:

In attributes, I placed the acf color field, and in its settings, in

after I filled it in like this:
style|background-color:

Before: !important

I hope it works for you too! Good luck.

![image](https://github.com/elementor/elementor/assets/19634278/028aadc9-39fa-4079-bce4-0235bb7a60a4)

**Comment by CloeSchorderet** [comment]
+1  it's not even a gradient in my case, just a an ACF background-color applied to squares. It should display the client's main colors in my projects.

**Comment by dribgib** [comment]
+1 -- this just started happening to me -- how is' there not a small snippet to catch this error .... 3 years??

**Comment by JoelEadeDesign** [comment]
I'm experiencing this as well on a dev site on Local. I've created dozens of Elementor sites with the only difference with this and others for me is that I enabled Beta mode to test out the beta Off Canvas element. It wasn't suitable so I rolled back to the normal Elementor & Elementor Pro versions. 

Not ideal but I can hide the warnings using the below:

`ini_set('display_errors','Off');
ini_set('error_reporting', E_ALL );
define('WP_DEBUG', false);
define('WP_DEBUG_DISPLAY', false);`

**Comment by pearllemon** [comment]
4 Years and 54 comments later and there is still no permanent solution to this issue. 
![image](https://github.com/user-attachments/assets/a53ffcdd-441a-40bb-babf-4225078c53a0)

**Comment by extrakerned** [comment]
Reporting this issue still exists on 3.23.4 / Pro 3.23.3

`Warning
: Undefined array key "background_color" in
.../wp-content/plugins/elementor/includes/conditions.php
on line
87
`

**Comment by GT-PaperDragon** [comment]
Although not a coded solution, I think I have found a workaround...

By adding a Background Overlay to the container and specifying a colour, the missing element to the array is completed. Set an Opacity of zero and this colour will not interfere with your dynamic/gradient colours.

Interested to hear if this works for others too.

![wp_temp](https://github.com/user-attachments/assets/78ed2201-d1d9-4abe-91ac-09ab129b1039)

**Comment by Bentonabento** [comment]
![full-3](https://github.com/user-attachments/assets/efed1494-be30-4838-91a5-83cb2b76bd2a)

Me, waiting for this issue to be solved since early 2020

**Comment by natequreshi** [comment]
Never solved !!!!!, just need need get it done with JetEngine, what I know with jetengine everything is possible except this particular job.

**Comment by toineenzo** [comment]
@nicholaszein Can you give an update on this? This bug has been reported 4 years ago! And for such a simple feature to not work, it should've been fixed already!

**Comment by Brayam** [comment]
> To achieve a gradient with a ACF dynamic color (`background-image: linear-gradient(180deg, transparent 0%, acf_color_field 75%);`), this worked for me.
> 
> 1. I set the Advanced > Attribute dynamically to the ACF color field I need (`acf_color_field`).
> 2. In the Attribute > Advanced > Before, I put this: `style|background-image: linear-gradient(180deg, transparent 0%, `
> 3. In the After, I put this:  `75%);`
> 
> Make sure there extra space at the end of #2 and the beginning of #2 so the CSS syntax is correct.
> 
> Here are the Attribute settings: ![2024-10-31_12-38-12](https://private-user-images.githubusercontent.com/44705874/382094655-15dc5cea-e778-49c6-93f0-9497492854f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2OTI1ODAsIm5iZiI6MTczMDY5MjI4MCwicGF0aCI6Ii80NDcwNTg3NC8zODIwOTQ2NTUtMTVkYzVjZWEtZTc3OC00OWM2LTkzZjAtOTQ5NzQ5Mjg1NGY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA0VDAzNTEyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMGI4Yzk1YjBiODkzMjRmYTI2NmIyODU3N2NlMzYwZmY0NTFjNDliYzNjZWIyNGU1MWRhMGQzMTE4MjAxZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y0sFFUMqXrHeFBjvLkuTeFHZkOO9nkcRZ_h-g0GONNI)
> 
> Here is the result: ![2024-10-31_12-45-46](https://private-user-images.githubusercontent.com/44705874/382094688-1f6ec637-0387-426f-b491-783638974571.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2OTI1ODAsIm5iZiI6MTczMDY5MjI4MCwicGF0aCI6Ii80NDcwNTg3NC8zODIwOTQ2ODgtMWY2ZWM2MzctMDM4Ny00MjZmLWI0OTEtNzgzNjM4OTc0NTcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA0VDAzNTEyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmNDY4NjIzNDlmY2I5MmI0MjBjNzIzOWQ5NGE4ZjlhNzBhZWM0MjNhM2IyNWYwZjcwZDM2MzY4YTI1ZTZmNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bmnivkqkkAzJQ5WyPUxYE4H_znXofBj1WHMNS4WlgWE)

Thanks, its works well. xD

**Comment by RezaSeifoori** [comment]
This is how I solved the problem
1 Enter the following code in the code snippet

add_action('wp_head', function () {
    if (is_singular()) { // فقط برای صفحات پست تایپ تکی اعمال می‌شود
        $color1 = get_post_meta(get_the_ID(), 'your_color_field_1', true);
        $color2 = get_post_meta(get_the_ID(), 'your_color_field_2', true);

        if (!empty($color1) && !empty($color2)) {
            echo "<style>:root { --color1: {$color1}; --color2: {$color2}; }</style>";
        }
    }
});
_your_color_field_1 and your_color_field_2 are the values ​​you chose for the Jet Engine color selection field name._
Then in Elementor Pro, in the Custom CSS section, you would add the following value to your container:
selector {
background-image: linear-gradient(148deg, var(--color1) 0%, var(--color2) 71%) !important;
}
